lib/commons/color/get-background-stack.js

```diff
@@ -2,59 +2,45 @@ import filteredRectStack from './filtered-rect-stack';
 import elementHasImage from './element-has-image';
 import getOwnBackgroundColor from './get-own-background-color';
 import incompleteData from './incomplete-data';
-import shadowElementsFromPoint from '../dom/shadow-elements-from-point';
 import reduceToElementsBelowFloating from '../dom/reduce-to-elements-below-floating';
 
 /**
- * Determines overlap of node's content with a bgNode. Used for inline elements
+ * Determine if element B is an inline descendant of A
  * @private
- * @param {Element} targetElement
- * @param {Element} bgNode
+ * @param {Element} node
+ * @param {Element} descendant
  * @return {Boolean}
  */
-function contentOverlapping(targetElement, bgNode) {
-  // get content box of target element
-  // check to see if the current bgNode is overlapping
-  var targetRect = targetElement.getClientRects()[0];
-  var obscuringElements = shadowElementsFromPoint(
-    targetRect.left,
-    targetRect.top
-  );
-  if (obscuringElements) {
-    for (var i = 0; i < obscuringElements.length; i++) {
-      if (
-        obscuringElements[i] !== targetElement &&
-        obscuringElements[i] === bgNode
-      ) {
-        return true;
-      }
-    }
+function isInlineDescendant(node, descendant) {
+  const CONTAINED_BY = Node.DOCUMENT_POSITION_CONTAINED_BY;
+  // eslint-disable-next-line no-bitwise
+  if (!(node.compareDocumentPosition(descendant) & CONTAINED_BY)) {
+    return false;
   }
-  return false;
+  const style = window.getComputedStyle(descendant)
+  const display = style.getPropertyValue('display');
+  return display.includes('inline')
 }
 
 /**
- * Calculate alpha transparency of a background element obscuring the current node
+ * Determine if the element obscures / overlaps with the text
  * @private
  * @param {Number} elmIndex
  * @param {Array} elmStack
  * @param {Element} originalElm
  * @return {Number|undefined}
  */
 function calculateObscuringElement(elmIndex, elmStack, originalElm) {
-  if (elmIndex > 0) {
-    // there are elements above our element, check if they contribute to the background
-    for (var i = elmIndex - 1; i >= 0; i--) {
-      const bgElm = elmStack[i];
-      if (contentOverlapping(originalElm, bgElm)) {
-        return true;
-      } else {
-        // remove elements not contributing to the background
-        elmStack.splice(i, 1);
-      }
+  // Reverse order, so that we can safely splice
+  for (let i = elmIndex - 1; i >= 0; i--) {
+    if (!isInlineDescendant(originalElm, elmStack[i])) {
+      return true;
     }
+    // Ignore inline descendants, for example:
+    // <p>text <img></p>; We don't care about the <img> element,
```

Comment:
What happens if `<img>` is positioned somehow, acting like a background?

Reply:
position: absolute / sticky force an element into display:block. I'll put in a test for it though.